blocks/header/header.js

```diff
@@ -1,170 +1,201 @@
+      if (MQ.matches) {
+        navSection.addEventListener('mouseover', mouseOverMenu.bind(a, block.parentNode));
       }
+      navSection.append(a);
+      navSection.append(icon);
+      return navSection;
     });
-  } else {
-    navDrops.forEach((drop) => {
-      drop.removeAttribute('role');
-      drop.removeAttribute('tabindex');
-      drop.removeEventListener('focus', focusNavSection);
+    // write the section menu
+    if (sectionMenus.length) {
+      sectionList.append(...sectionMenus);
+    }
+
+    // add event listeners
+    // for desktop when clicking anywhere on the document
+    document.addEventListener('click', closeMenuSection);
+    document.addEventListener('mouseover', closeMenuSection);
+
+    // for the hamburger toggle icon
+    nav.querySelector('.hamburger-toggle').addEventListener('click', (e) => {
+      e.preventDefault();
+      const isExpanded = e.currentTarget.getAttribute('aria-expanded') === 'true';
+      if (isExpanded) {
+        e.currentTarget.setAttribute('aria-expanded', 'false');
+        document.querySelector('header nav').setAttribute('aria-expanded', 'false');
+      } else {
+        e.currentTarget.setAttribute('aria-expanded', 'true');
+        document.querySelector('header nav').setAttribute('aria-expanded', 'true');
+      }
     });
-  }
-  // enable menu collapse on escape keypress
-  if (!expanded || isDesktop.matches) {
-    // collapse menu on escape press
-    window.addEventListener('keydown', closeOnEscape);
-  } else {
-    window.removeEventListener('keydown', closeOnEscape);
-  }
-}
 
-function linkBrand(nav) {
-  const navBrand = nav.querySelector('.nav-brand');
-  const homeLink = document.createElement('a');
-  homeLink.href = '/';
-  homeLink.ariaLabel = 'Go to Home';
-  homeLink.append(...navBrand.childNodes);
-  navBrand.append(homeLink);
-}
+    // for the search toggle icon
+    nav.querySelector('.search-label').addEventListener('click', (e) => {
+      e.preventDefault();
+      const isExpanded = e.currentTarget.getAttribute('aria-expanded') === 'true';
+      if (isExpanded) {
+        e.currentTarget.setAttribute('aria-expanded', 'false');
+        document.querySelector('header nav .search .search-container').setAttribute('aria-expanded', 'false');
+      } else {
+        e.currentTarget.setAttribute('aria-expanded', 'true');
+        document.querySelector('header nav .search .search-container').setAttribute('aria-expanded', 'true');
+      }
+    });
+
+    // force hamburger close when in destop size
+    MQ.addEventListener('change', () => {
+      document.querySelector('header nav .hamburger-toggle').setAttribute('aria-expanded', 'false');
+      document.querySelectorAll('header nav .sections .expand').forEach((el) => el.classList.remove('expand'));
+      // remove hard styled heights (from animations)
+      document.querySelectorAll('header nav .sections ul').forEach((ul) => {
+        ul.style.height = null;
```

Comment:
unset instead of null may work.

Reply:
That does not work in JS. It is only in CSS.
Setting it to an empty string is more appropriate.